<?xml version="1.0" encoding="UTF-8"?>
<source package="org.apache.commons.math3.exception">
  <import name="org.apache.commons.math3.exception.util.Localizable" />
  <import name="org.apache.commons.math3.exception.util.ExceptionContext" />
  <import name="org.apache.commons.math3.exception.util.ExceptionContextProvider" />
  <class name="MathRuntimeException" extends="RuntimeException" startLine="23">
    <implements name="ExceptionContextProvider" />
    <javadoc>
      <text>* As of release 4.0, all exceptions thrown by the Commons Math code (except{@link NullArgumentException}) inherit from this class.
 * In most cases, this class should not be instantiated directly: it should
 * serve as a base class for implementing exception classes that describe a
 * specific "problem".</text>
      <since>3.1</since>
      <version>$Id: MathRuntimeException.java 1416643 2012-12-03 19:37:14Z tn $</version>
    </javadoc>
    <javadoc>
      <text>* Serializable version Id.</text>
    </javadoc>
    <declaration type="long" name="serialVersionUID" />
    <javadoc>
      <text>* Context.</text>
    </javadoc>
    <declaration type="ExceptionContext" name="context" />
    <javadoc>
      <param>pattern Message pattern explaining the cause of the error.</param>
      <param>args Arguments.</param>
    </javadoc>
    <method type="constructor" name="MathRuntimeException" startLine="45" endLine="48" />
    <javadoc>
      <text>* {@inheritDoc}</text>
    </javadoc>
    <method type="ExceptionContext" name="getContext" startLine="51" endLine="53" />
    <javadoc>
      <text>* {@inheritDoc}</text>
    </javadoc>
    <method type="String" name="getMessage" startLine="57" endLine="59" />
    <javadoc>
      <text>* {@inheritDoc}</text>
    </javadoc>
    <method type="String" name="getLocalizedMessage" startLine="63" endLine="65" />
  </class>
</source>
